crypto/CMakeLists.txt

```diff
@@ -96,15 +93,15 @@ if (CONFIG_NRF_CC310_BL)
   target_link_libraries(nrfxlib_crypto INTERFACE ${CC310_BL_LIB})
 endif()
 
-if (CONFIG_NRF_CC3XX_PLATFORM)
-  if(CONFIG_SOC_NRF5340_CPUAPP)
-    set (CC3XX_ARCH cc312)
-  else()
-    set (CC3XX_ARCH cc310)
-  endif()
+if(CONFIG_SOC_NRF5340_CPUAPP)
+  set(CC3XX_ARCH cc312)
+else()
+  set(CC3XX_ARCH cc310)
```

Comment:
This change is not really a refactor, CC3XX_ARCH will be more broadly defined by this change.

I think instead of else doing `if (CONFIG_HAS_HW_NRF_CC310)` is a more logical check (or elseif if you prefer).
then we don't end up with CC3XX_ARCH being defined as cc310 on non-cryptocell devices, or possibly being the default when a new cc3xx arch is added.